src/main/resources/db/migration/all/20240521110000__create_live_view.sql

```diff
@@ -0,0 +1,48 @@
+-- replace application summary views
+CREATE OR REPLACE VIEW cas_2_application_summary AS SELECT
+    CAST(a.id AS TEXT),
+    a.crn,
+    a.noms_number,
+    CAST(a.created_by_user_id AS TEXT),
+    nu.name,
+    a.created_at,
+    a.submitted_at,
+    a.hdc_eligibility_date,
+    asu.label,
+    CAST(asu.status_id AS TEXT),
+    a.referring_prison_code,
+    a.conditional_release_date,
+    asu.created_at AS status_created_at
+FROM cas_2_applications a
+LEFT JOIN (SELECT DISTINCT ON (application_id) su.application_id, su.label, su.status_id, su.created_at
+    FROM cas_2_status_updates su
+    ORDER BY su.application_id, su.created_at DESC) as asu
+    ON a.id = asu.application_id
+JOIN nomis_users nu ON nu.id = a.created_by_user_id;
+
+CREATE OR REPLACE VIEW cas_2_application_live_summary AS SELECT
+    a.id,
+    a.crn,
+    a.noms_number,
+    a.created_by_user_id,
+    a.name,
+    a.created_at,
+    a.submitted_at,
+    a.hdc_eligibility_date,
+    a.label,
+    a.status_id,
+    a.referring_prison_code
+FROM cas_2_application_summary a
+WHERE (a.conditional_release_date IS NULL OR a.conditional_release_date >= current_date)
+AND (a.label IS NULL
+    OR (a.label = 'Referral withdrawn' AND a.status_created_at > (current_date - INTERVAL '14 DAY'))
```

Comment:
I'm a bit worried that this is creating a dependency on the label for a couple of reasons - 
1. If we were to add a new type of status update would it be incorrectly hidden in this case? It wouldn't be clear to me that I would need to update this view query whenever a new status update is added. Is there a way of having the query only affect those three that we know have the 14 day cut off?
2. The label is the human readable label for the status update which could change, I think it should be based on the `status_id` which is the static ID for the type of status update (see `Cas2ApplicationStatusSeeding`). This will make it unreadable though so might need some comments 😬

Reply:
Good points.  Thanks!

Ref 1. I have changed the logic from IN LIST () to NOT IN LIST () so this addresses this point.

Ref 2. I was conscious it wasn't great comparing strings in the original SQL so comparing by UUID is better, I've made this change and have added a comment to say which UUID corresponds to which status.  Maybe at some point in the future these statuses will go into a dedicated Postgres table.